backend/src/main/java/com/skr/virtuallibrary/services/UserService.java

```diff
@@ -63,6 +64,18 @@ public UnregisteredUser findUnregisteredUserByToken(String token) {
                 .orElseThrow(() -> new UserNotFoundException("User could not be found with token: " + token));
     }
 
+    public UserDto updateUser(UpdateUserRequest request) {
+        User user = getCurrentUser();
+        user.setFirstName(request.getFirstName());
+        user.setLastName(request.getLastName());
+        user.setEmail(request.getEmail());
+        user.setLanguage(request.getLanguage());
+        if (request.getProfilePictureId() != null) {
```

Comment:
Tu powinna być większa walidacja. Skoro `profilePicture` w Controllerze nie jest obowiązkowe, to ktoś może wysłać requesta z błędnym id zdjęcia profilowego. Serwis powinien wiedzieć czy zdjęcie zostało nadesłane z Controllera lub sprawdzić, czy takowe już istnieje.